prow/tide/tide_test.go

```diff
@@ -3942,3 +3951,74 @@ func TestDeduplicateContestsDoesntLoseData(t *testing.T) {
 		})
 	}
 }
+
+func TestPickSmallestPassingNumber(t *testing.T) {
+	priorities := []config.TidePriority{
+		{Labels: []string{"kind/failing-test"}},
+		{Labels: []string{"area/deflake"}},
+		{Labels: []string{"kind/bug", "priority/critical-urgent"}},
+	}
+	testCases := []struct {
+		name     string
+		prs      []PullRequest
+		expected int
+	}{
+		{
+			name: "no label",
+			prs: []PullRequest{
+				testPR("org", "repo", "A", 5, githubql.MergeableStateMergeable),
+				testPR("org", "repo", "A", 3, githubql.MergeableStateMergeable),
+			},
+			expected: 3,
+		},
+		{
+			name: "deflake PR",
+			prs: []PullRequest{
+				testPR("org", "repo", "A", 5, githubql.MergeableStateMergeable),
+				testPR("org", "repo", "A", 3, githubql.MergeableStateMergeable),
+				testPRWithLabels("org", "repo", "A", 7, githubql.MergeableStateMergeable, []string{"area/deflake"}),
+			},
+			expected: 7,
+		},
+		{
+			name: "same label",
+			prs: []PullRequest{
+				testPRWithLabels("org", "repo", "A", 7, githubql.MergeableStateMergeable, []string{"area/deflake"}),
+				testPRWithLabels("org", "repo", "A", 6, githubql.MergeableStateMergeable, []string{"area/deflake"}),
+				testPRWithLabels("org", "repo", "A", 1, githubql.MergeableStateMergeable, []string{"area/deflake"}),
+			},
+			expected: 1,
+		},
+		{
+			name: "missing one label",
+			prs: []PullRequest{
+				testPR("org", "repo", "A", 5, githubql.MergeableStateMergeable),
+				testPR("org", "repo", "A", 3, githubql.MergeableStateMergeable),
+				testPRWithLabels("org", "repo", "A", 6, githubql.MergeableStateMergeable, []string{"kind/bug"}),
+			},
+			expected: 3,
+		},
+		{
+			name: "complete",
+			prs: []PullRequest{
+				testPR("org", "repo", "A", 5, githubql.MergeableStateMergeable),
+				testPR("org", "repo", "A", 3, githubql.MergeableStateMergeable),
+				testPRWithLabels("org", "repo", "A", 6, githubql.MergeableStateMergeable, []string{"kind/bug"}),
+				testPRWithLabels("org", "repo", "A", 7, githubql.MergeableStateMergeable, []string{"area/deflake"}),
+				testPRWithLabels("org", "repo", "A", 8, githubql.MergeableStateMergeable, []string{"kind/bug"}),
+				testPRWithLabels("org", "repo", "A", 9, githubql.MergeableStateMergeable, []string{"kind/failing-test"}),
+				testPRWithLabels("org", "repo", "A", 10, githubql.MergeableStateMergeable, []string{"kind/bug", "priority/critical-urgent"}),
+			},
+			expected: 9,
+		},
+	}
+	alwaysTrue := func(*logrus.Entry, githubClient, PullRequest, contextChecker) bool { return true }
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			_, got := pickHighestPriorityPR(nil, nil, tc.prs, nil, alwaysTrue, priorities)
```

Comment:
@alvaroaleman for easier unit tests, I have moved the func `isPassingTests` to a parameter that I can mock more easily than the reall one